python_hb_report/collector.py

```diff
@@ -0,0 +1,374 @@
+		msg = msg + rsag_ver
+
+		crm_version = utillib.crm_info()
+		msg = msg+crm_version
+
+		booth_info = utillib.do_command(['booth','--version'])
+		msg = msg+booth_info
+		
+		pkg_info = utillib.pkg_version()
+		msg = msg + pkg_info
+		f.write(pkg_info)
+		
+		if envir.SKIP_LVL >= 1:
+			vrf_info = utillib.verify_packages()
+			msg = msg + vrf_info
+
+		sys_name = 'Platform: '+ platform.system()+'\n'
+		msg = msg+sys_name
+
+		knl_name = 'Kernel release: '+platform.release()+'\n'
+		msg = msg+knl_name
+		
+		arch_name = 'Architecture: '+platform.machine()+'\n'
+		msg = msg+arch_name
+
+		if platform.system() == 'Linux':
+			dist_name = utillib.distro()+'\n'
+			msg = msg + dist_name
+		f.write(msg)
+		f.close()
+
+	def sys_stats(self):
+		
+		msg = ''
+
+		f = open(os.path.join(self.WORKDIR,envir.SYSSTATS_F),'w')
+
+		msg = msg + self.WE+'\n'
+
+		uptime = utillib.do_command(['uptime'])
+		msg = msg+uptime
+
+		ps_info = utillib.do_command(['ps','axf'])
+		msg = msg+ps_info
+	
+		ps_info = utillib.do_command(['ps','auxw'])
+		msg = msg + ps_info
+
+		top_info = utillib.do_command(['top','-b','-n','1'])
+		msg = msg + top_info
+
+		ip_info = utillib.do_command(['ip','addr'])
+		msg = msg+'\n'+ip_info
+
+		net_info = utillib.do_command(['netstat','-i'])
+		msg = msg +'\n'+net_info
+
+		arp_info = utillib.do_command(['arp','-an'])
+		msg = msg + '\n' + arp_info
+
+		if os.path.isdir('/proc'):
+			cpu_f =open('/proc/cpuinfo','r')
+			cpu_info = cpu_f.readline()
+			while len(cpu_info):
+				msg = msg + cpu_info
+				cpu_info = cpu_f.readline()
+
+		scsi_info = utillib.do_command(['lsscsi'])
+		msg = msg +'\n'+ scsi_info
+
+		pci_info = utillib.do_command(['lspci'])
+		msg = msg +'\n'+ pci_info
+
+		mount_info = utillib.do_command(['mount'])
+		msg = msg +'\n' + mount_info
+
+		#df can block, run in background, allow for 5 seconds
+		df_pro = subprocess.Popen(['df'],stderr = subprocess.STDOUT,stdout = subprocess.PIPE)
+		timer = threading.Timer(5.0,df_pro.kill)
+		timer.start()
+		df_info = df_pro.communicate()[0]
+		if timer.is_alive():
+			#df exited naturally, cancel timer
+			timer.cancel()
+		msg = msg +'\n'+df_info
+
+		f.write(msg)
+		f.close()
+
+	def pe2dot(self,path):
+		pef = utillib.basename(path)
+		if pef.endswith('.bz2'):
+			dotf = pef[0:len(pef)-4]
+
+		if not len(envir.PTEST):
+			return False
+		try:
+			msg = utillib.do_command([envir.PTEST,'-D','dotf','-x',pef])
+		except:
+			utillib.debug(envir.PTEST+' faild! ')
+			return
+
+	def dlm_dump(self):
+		output = os.path.join(self.WORKDIR,envir.DLM_DUMP_F)
+		if not utillib.do_which('dlm_tool'):
+			return 
+		dlm_file_info = 'NOTICE - Lockspace overview:\n'	
+		dlm_file_info = dlm_file_info + utillib.do_command(['dlm_tool','ls'])
+
+		dlm_pro = subprocess.Popen(['dlm_tool','ls'],stdout = subprocess.PIPE,stderr = subprocess.STDOUT)
+		grep_pro = subprocess.Popen(['grep','name'],stdin = dlm_pro.stdout,stdout = subprocess.PIPE,stderr = subprocess.STDOUT)
```

Comment:
in line 145, you 've called dlm_tool ls, and in line 147, you call it again, why?